pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -1363,6 +1365,42 @@ func (c *K8sOrchestrator) GetVolumeAttachment(ctx context.Context, volumeId stri
 	return volumeAttachment, nil
 }
 
+// GetNodeForAttachedVolume returns the node name on which a Pod is running for a given volumeID.
+func (c *K8sOrchestrator) GetNodeForAttachedVolume(ctx context.Context, volumeHandle string) (string, error) {
+	log := logger.GetLogger(ctx)
+	var namespacedPvcName string
+	for volumeID, pvc := range c.volumeIDToPvcMap.items {
+		if volumeID == volumeHandle {
+			namespacedPvcName = pvc
+		}
+	}
+	pvcNameNamespaceSplit := strings.Split(namespacedPvcName, "/")
+	pods, err := c.k8sClient.CoreV1().Pods(pvcNameNamespaceSplit[0]).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		log.Errorf("Failed to get the list of pods in namespace: %q. Err: %v", pvcNameNamespaceSplit[0], err)
+		return "", err
+	}
+	pvc, err := c.k8sClient.CoreV1().PersistentVolumeClaims(pvcNameNamespaceSplit[0]).Get(
+		ctx, pvcNameNamespaceSplit[1], metav1.GetOptions{})
+	if err != nil {
+		log.Errorf("Failed to get the pvc %q in namespace: %q. Err: %v",
+			pvcNameNamespaceSplit[1], pvcNameNamespaceSplit[0], err)
+		return "", err
+	}
+	if *pvc.Spec.VolumeMode == v1.PersistentVolumeBlock {
+		// Identify the pod that has a volume with name "pvcName" associated with it.
+		for _, pod := range pods.Items {
+			for _, volume := range pod.Spec.Volumes {
+				pvClaim := volume.VolumeSource.PersistentVolumeClaim
+				if pvClaim != nil && pvClaim.ClaimName == pvcNameNamespaceSplit[1] {
+					return pod.Spec.NodeName, nil
+				}
+			}
+		}
+	}
+	return "", errors.New("")
```

Comment:
May be log an error message before returning that no matching node is found? The same message string can be passed to errors.New().

Reply:
Removed as it is not needed